lib/Sema/CSRanking.cpp

```diff
@@ -1424,63 +1431,15 @@ ConstraintSystem::findBestSolution(SmallVectorImpl<Solution> &viable,
     return bestIdx;
   }
 
-  // If there is not a single "better" than others
-  // solution, which probably means that solutions
-  // were incomparable, let's just keep the original
-  // list instead of removing everything, even if we
-  // are asked to "minimize" the result.
-  if (losers.size() == viable.size())
-    return None;
-
-  // The comparison was ambiguous. Identify any solutions that are worse than
-  // any other solution.
-  for (unsigned i = 0, n = viable.size(); i != n; ++i) {
-    // If the first solution has already lost once, don't bother looking
-    // further.
-    if (losers[i])
-      continue;
-
-    for (unsigned j = i + 1; j != n; ++j) {
-      // If the second solution has already lost once, don't bother looking
-      // further.
-      if (losers[j])
-        continue;
-
-      switch (compareSolutions(*this, viable, diff, i, j)) {
-      case SolutionCompareResult::Identical:
-        // FIXME: Dub one of these the loser arbitrarily?
-        break;
-
-      case SolutionCompareResult::Better:
-        losers[j] = true;
-        break;
-
-      case SolutionCompareResult::Worse:
-        losers[i] = true;
-        break;
-
-      case SolutionCompareResult::Incomparable:
-        break;
-      }
-    }
-  }
-
-  // Remove any solution that is worse than some other solution.
-  unsigned outIndex = 0;
-  for (unsigned i = 0, n = viable.size(); i != n; ++i) {
-    // Skip over the losing solutions.
-    if (losers[i])
-      continue;
-
-    // If we have skipped any solutions, move this solution into the next
-    // open position.
-    if (outIndex < i)
-      viable[outIndex] = std::move(viable[i]);
-
-    ++outIndex;
+  // If mimization is requested, let's remove all of the solutions that
+  // score higher than the set of best ones.
+  if (minimize) {
+    viable.erase(llvm::remove_if(viable,
+                                 [&bestScore](const Solution &solution) {
+                                   return solution.getFixedScore() > bestScore;
+                                 }),
+                 viable.end());
   }
-  viable.erase(viable.begin() + outIndex, viable.end());
-  NumDiscardedSolutions += viable.size() - outIndex;
```

Comment:
@xedin Just a question, does `NumDiscardedSolutions` still need to be update since some solutions in `viable` are still being erased and if that could have some effect later(not sure where this is used)?

Reply:
It's just a perf counter, and I'm not even sure it's a useful one... I'm just trying to see whether this broke anything and then I'm going to deal with that...

Reply:
No need to apologize! Please feel free to ask questions when something is unclear :)